src/SFML/Graphics/Font.cpp

```diff
@@ -495,9 +517,12 @@ void Font::setSmooth(bool smooth)
     {
         m_isSmooth = smooth;
```

Comment:
I'm not sure what you mean by "involved". If it's to avoid the extra level of scope I'm not sure it's worth it. The function is only a dozen lines and 3 levels of scope

Reply:
I meant as an optimization, to not run the rest of the function if nothing changes.

[It's what we already do in `sf::Texture::setSmooth()`.](https://github.com/SFML/SFML/blob/master/src/SFML/Graphics/Texture.cpp#L643) Having an extra indentation is not great though, hence suggestion of early return.